### PR TITLE
Fix resolving tasks if original task was delegated

### LINKED ISSUE #4230: TaskTransitionController.all_subtasks_finished does not handle successor pairs.
The current [`all_subtasks_finished`](https://github.com/4teamwork/opengever.core/blob/master/opengever/task/browser/transitioncontroller.py#L565) check in the transition controller, currently only checks for subtasks of the current task but it should also check for subtasks on the successor or predecessor if there are some. 

Otherwise it can lead to confusing failures when trying to resolve a task on the predecessor site, with subtasks on the successor site. 

https://extranet.4teamwork.ch/extern/opengever-kanton-zug/sprint-backlog/20377

### PR DESCRIPTION
**Pull request by tarnap**
## Assuming

1. Anton created a successor task for Berta
2. Anton delegated the task to Carlos

```
FD                   RK

              1.
Task (Anton) +-----> Successor Task (Berta)
 +
 | 2.
 | 
 v
Delegated Task (Carlos)

```

When Berta tries to resolve the task, the status of the subtask delegated to Carlos should be checked as well.

Therefore the check on Berta's side should get all the subtasks from
Berta's and Anton's task.

Resolves #4230 
Closes #4351

### CODE REVIEW

**Comment by tarnap** [comment]
@phgross: This PR is ready for review

File: opengever/task/tests/test_task_checker.py [hunk 1]
```diff
@@ -1,199 +1,132 @@
...
-
-        task1 = create(Builder('task').having(responsible_client='client1'))
-
+        self.login(self.secretariat_user)
         self.assertTrue(
-            get_checker(task1).current_user.in_responsible_orgunits_inbox_group)
+            get_checker(self.task).current_user.in_responsible_orgunits_inbox_group)
 
-        checker = Checker(task1.get_sql_object(), self.request, self.hugo)
+        checker = Checker(self.task.get_sql_object(), self.request, self.secretariat_user)
```
Comment (phgross) [inline opengever/task/tests/test_task_checker.py]: I would add a short comment that one check is with `agency` and one without ...

File: opengever/task/tests/test_task_checker.py [hunk 2]
```diff
@@ -1,199 +1,132 @@
...
-            .successor_from(predecessor)
-            .in_state('task-state-in-progress')
-            )
+        self.set_workflow_state('task-state-in-progress', self.subtask)
+        self.set_workflow_state('task-state-in-progress', self.archive_task)
+        self.register_successor(self.task, self.archive_task)
 
         sql_successor_task = ISuccessorTaskController(
-            predecessor).get_successors()[0]
+            self.task).get_successors()[0]
```
Comment (phgross) [inline opengever/task/tests/test_task_checker.py]: fetching the sql_successor_task should not be necessary its `self.archive_task`

**Comment by tarnap** [comment]
@phgross: please re-re-re-review this PR 🙈

**Pull request merged**